### ISSUE TITLE
Force k3s update

### ISSUE THREAD

**Issue by nicolaballotta** [kube-hetzner/terraform-hcloud-kube-hetzner#562]
Is there a way to force k3s update?

I've my HA cluster and k3s update probably had some issues, since it's enabled but didn't update everywhere.

```
++++-master-fsn1-flo   Ready    control-plane,etcd,master   65d   v1.24.9+k3s1    10.255.0.101   +++++++++++       openSUSE MicroOS   6.0.12-1-default   containerd://1.6.12-k3s1
++++-master-hel1-uro   Ready    control-plane,etcd,master   65d   v1.24.9+k3s1    10.253.0.101   +++++++++++       openSUSE MicroOS   6.0.12-1-default   containerd://1.6.12-k3s1
++++-master-nbg1-awj   Ready    control-plane,etcd,master   65d   v1.24.8+k3s1    10.254.0.101   +++++++++++       openSUSE MicroOS   6.0.12-1-default   containerd://1.6.8-k3s1
++++-worker-1-xsx      Ready    <none>                      65d   v1.24.8+k3s1    10.0.0.101     +++++++++++       openSUSE MicroOS   6.0.12-1-default   containerd://1.6.8-k3s1
++++-worker-2-amz      Ready    <none>                      65d   v1.24.8+k3s1    10.1.0.101     +++++++++++       openSUSE MicroOS   6.0.12-1-default   containerd://1.6.8-k3s1
++++-worker-3-sps      Ready    <none>                      76m   v1.24.10+k3s1   10.2.0.101     +++++++++++       openSUSE MicroOS   6.1.8-1-default    containerd://1.6.15-k3s1
```

Any clue?

**Comment by mysticaltech** [comment]
@nicolaballotta Some of this is explained in the readme. But here's the info too! https://milindchawre.github.io/site/blog/k3s-upgrade-a-simplified-guide/

Basically, you have to see what's going on with the system upgrade controller, you can do so with by running `kubectl get plan -n system-upgrade`, and then `kubectl describe`, each.

What I am think is, either the upgrade of k3s has been disabled with one of the methods mentioned in the readme. Or you actually have an old version that has a small problem in kured.

You can actually `kubectl get daemonset kured -n kube-system` and verify that at the bottom, none of these options are missing:

https://github.com/kube-hetzner/terraform-hcloud-kube-hetzner/blob/6481a9f1bdc423d7e75c8cb404eff06ae45bff13/locals.tf#L467-L470

Basically, if a node is stuck with the label `kured=rebooting` (you can check that by describing the node), the system upgrade controller plan that is in charge of upgrading that node will not kick in, as it thinks that kured is doing a reboot after an OS upgrades.

Check all of this, please!

**Comment by nicolaballotta** [comment]
@mysticaltech thank you for pointing me in the right direction, I missed that part of the documentation and yeah the problems were multiple. 

- I was missing one of the options you mentioned
- I had almost all nodes with `kured=rebooting`. In that case I think the problem was due to Longhorn not allowing to eviction of instance-manager (also if `Allow Node Drain with the Last Healthy Replica` is enabled). I followed the solution described [here](https://github.com/longhorn/longhorn/issues/4652) and I finally solved.

**Comment by mysticaltech** [comment]
Great to hear @nicolaballotta! Actually, the `kured=rebooting` was just us forgetting to include the missing kured option in a previous version of the module.  You need it to remove the "rebooting" label just after a successful reboot.
